src/core/thread/address_resolver.cpp

```diff
@@ -162,6 +162,34 @@ void AddressResolver::InvalidateCacheEntry(Cache &aEntry)
     otLogInfoArp(GetInstance(), "cache entry removed!");
 }
 
+void AddressResolver::UpdateCacheEntry(const Ip6::Address &aEid, Mac::ShortAddress aRloc16)
+{
+    for (int i = 0; i < kCacheEntries; i++)
+    {
+        if (mCache[i].mState == Cache::kStateInvalid ||
+            memcmp(&mCache[i].mTarget, &aEid, sizeof(mCache[i].mTarget)) != 0)
+        {
+            continue;
+        }
+
+        if (mCache[i].mRloc16 != aRloc16)
+        {
+            // not updating the age here is intentional because this cache entry is not actually being used
+            mCache[i].mRloc16 = aRloc16;
+            mCache[i].mRetryTimeout = 0;
+            mCache[i].mLastTransactionTime = 0;
+            mCache[i].mTimeout = 0;
+            mCache[i].mFailures = 0;
+            mCache[i].mState = Cache::kStateCached;
```

Comment:
Just checking that possibly not setting `mMeshLocalIid` is ok here....(if the cache state was `kStateQuery`)...